draft-ietf-quic-transport.md

```diff
@@ -3270,14 +3270,25 @@ delivery, as long as it is not in violation of the receiver's flow control
 limits.
 
 An endpoint MUST NOT send data on any stream without ensuring that it is within
-the data limits set by its peer.  The cryptographic handshake stream, Stream 0,
-is exempt from the connection-level data limits established by MAX_DATA. Data on
-stream 0 other than the initial cryptographic handshake message is still subject
-to stream-level data limits and MAX_STREAM_DATA. This message is exempt
-from flow control because it needs to be sent in a single packet regardless of
-the server's flow control state. This rule applies even for 0-RTT handshakes
-where the remembered value of MAX_STREAM_DATA would not permit sending a full
-initial cryptographic handshake message.
+the data limits set by its peer.
+
+The cryptographic handshake stream, Stream 0, is exempt from the
+connection-level data limits established by MAX_DATA.  That is, data sent on
+stream 0 is not counted against the limit expressed by MAX_DATA.  Data on stream
+0 is still subject to stream-level data limits and MAX_STREAM_DATA.
+
+Data sent in the Initial and Retry packets do not count toward stream 0 flow
+control limits.  These packet types do not permit the sending of data in
+multiple packets, so there is no opportunity for a peer to send MAX_STREAM_DATA
+frames.  Note also that the next packet sent after either of these packets might
+start at a stream offset of 0.  Thus, the only limit on the size of the
+cryptographic handshake messages these packets contain is determined by the MTU.
+
+Endpoints MUST respect the stream flow control limit set by their peer.  A
+client might exceed this limit when sending an Initial packet and so might need
```

Comment:
Actually, no, they can't -- ClientHello has to fit in the single Initial packet, which you say on 3280 doesn't count against this limit, so who cares?

A client might exceed this limit in a Handshake packet, if the handshake takes more than 1 RTT (or the Client Finished is huge for some reason, I suppose), though.  Is that what you're after?

Reply:
I should delete this sentence, I don't know where it came from.

Yes, I was looking for flow control for client certificates.

Reply:
I think if you change Initial to Handshake, this is fine.

Reply:
Yes, I noticed that when reworking this.